src/generate/generate.ts

```diff
@@ -0,0 +1,105 @@
+import assert from "node:assert";
+import { readFileSync, realpathSync, rmSync, writeFileSync } from "node:fs";
+import { basename, dirname, join } from "node:path";
+import type { AstroIntegration, AstroIntegrationLogger } from "astro";
+import { generate as generateCliOptionsMarkdown } from "./generate-cli-options";
+import { LYCHEE_VERSION } from "./lychee-version";
+
+class Generator {
+	constructor(
+		private logger: AstroIntegrationLogger,
+		private addWatchFile: (path: URL | string) => void,
+	) {}
+
+	// Fetch file from main lychee repository at the currently pinned tag
+	private async fetchFromRepository(filePath: string) {
+		this.logger.info(`Fetching ${filePath} from git tag ${LYCHEE_VERSION}`);
+		const url = `https://raw.githubusercontent.com/lycheeverse/lychee/refs/tags/${LYCHEE_VERSION}/${filePath}`;
+
+		const readme = await fetch(url);
+		assert(readme.ok, `${readme.status} when fetching ${url}`);
+		return readme.text();
+	}
+
+	// Rewrite the template file at templatePath to a non-template file
+	// and replace the placeholder in the process.
+	private async applyPlaceholder(
+		templatePath: string,
+		replacements: Array<{ placeholder: string; value: string }>,
+	) {
+		const [dir, file] = [dirname(templatePath), basename(templatePath)];
```

Comment:
You can use https://nodejs.org/api/path.html#pathparsepath which gives you a pretty little object with the path components.